### PR TITLE
I538 support interactive problems

### LINKED ISSUE #538: Add support for interactive problems
**Describe the issue**: 

Add support for interactive problems.

TODO - find specification reference for interactive problems

In the pretest contest ( https://github.com/icpcsysops/ccsconfig-dhaka/tree/master/contests/pretest) there is a 'interactive' identifier identifying that the problem is interactive (?)

config/accessdenied/problem.yaml: validation: custom interactive
config/boredombuster/problem.yaml: validation: custom interactive

**To Reproduce**: 

**Expected behavior**: 

**Actual behavior**: 

**Environment**: 

**Log Info**:  

**Screenshots**:  

**Additional context**: 

wf 2022 pretest had a interactive problem

### PR DESCRIPTION
**Pull request by johnbrvc**
### Description of what the PR does
Add built-in support for interactive problems.
Previously, interactive problems were handled by munging the execute command line from the edit problem GUI or in a problem's YAML file.
This PR makes PC2 aware of interactive problems.  Special custom shell scripts are used to setup the pipe work needed to connect a run's I/O to the O/I of a CLICS compliant custom interactive validator.  There are separate scripts for sandbox and non-sandbox problems.
An additional special script is used during the PC2 `validateProgram()` phase that copies the results of an interactive run performed by one of the aforementioned shell scripts to the location desired by PC2.

Summary of changes to the code:

- Add support in `ContestSnakeYAMLLoader.java` to recognize an interactive problem and perform appropriate validation of the validator supplied.
- Add support to the `ProblemPane` GUI to indicate if a problem is interactive.
- Add support to the `EditProblemPane `GUI to allow configuration of interactive problems.
- Add support for interactive problems to the `Problem` object.
- Expandthe `CustomValidatorSettings `object to be aware of interactive problems.
- `Executable.java`

  - Change to deal with interactive problems.
  - Refactor some code to put commonly used blocks in methods.
  - Add more substitution variables to support interactive problems and also allow for better logging in validators (`{:testcase}`, `{:executeinfofilenam}`) .
  - Add routines to set up the interactive environment in the execute folder.

- Create new scripts folder to hold all helper scripts used by sandboxes and interactive validators.  This involved getting rid of the `sandbox `folder (really, renaming it) and creating the `scripts `folder. 
- CI: Fix missing "`toString`" properties.
- CI: Make` teams2.tsv` be the preferred teams file during loading of a CDP.  This provides organizations.
- CI: better reporting of errors during reading of the CDP when loading a contest.  Combats the antagonizing "_check logs_" message previously reported.
- CI: Update copyright notices

### Issue which the PR addresses
Fixes #538

### Environment in which the PR was developed (OS,IDE, Java version, etc.)
Windows 11/Ubuntu 22.04.2
Eclipse
java version "1.8.0_321"
Java(TM) SE Runtime Environment (build 1.8.0_321-b07)
Java HotSpot(TM) 64-Bit Server VM (build 25.321-b07, mixed mode)

### Precise steps for _testing_ the PR 
Coming soon.

### CODE REVIEW

File: scripts/pc2_interactive.sh [hunk 1]
```diff
@@ -0,0 +1,311 @@
...
+
+usage()
+{
+  cat <<SAGE
+Usage: pc2_interactive.sh interactive-validator judgein judgeans testcase command command_args
+
+interactive-validator, judge's program to execute to provide interactive responses to the submission
+judgein, judges input file
+judgeans, judges answer file
+testcase, test case number (for logging)
```
Comment (lane55) [inline scripts/pc2_interactive.sh:89]: missing command and command_args descriptions

File: scripts/pc2_interactive.sh [hunk 2]
```diff
@@ -0,0 +1,311 @@
...
+<?xml version="1.0"?>
+<result outcome="$msg1" security="$INT_RESULTFILE">$msg2</result>
+EOF
+	sent_xml=1
+}
+
+# ------------------------------------------------------------
+
+if [ "$#" -lt 1 ] ; then
+   echo $0: No command line arguments
```
Comment (lane55) [inline scripts/pc2_interactive.sh]: Missing commnd line...

File: scripts/pc2sandbox_interactive.sh [hunk 3]
```diff
@@ -0,0 +1,448 @@
...
+{
+  cat <<SAGE
+Usage: pc2sandbox_interactive.sh memlimit timelimit interactive-validator judgein judgeans testcase command command_args
+
+memlimit, in MB
+timelimit, in seconds
+interactive-validator, judge's program to execute to provide interactive responses to the submission
+judgein, judges input file
+judgeans, judges answer file
+testcase, test case number (for logging)
```
Comment (lane55) [inline scripts/pc2sandbox_interactive.sh:114]: Add command and command_args description

File: src/edu/csus/ecs/pc2/core/Constants.java [hunk 4]
```diff
@@ -188,13 +188,40 @@ private Constants() {
...
      * Sandbox constants
      */
     public static final String PC2_INTERNAL_SANDBOX_COMMAND_LINE = "./{:sandboxprogramname} {:memlimit} {:timelimit}";
 
     public static final String PC2_INTERNAL_SANDBOX_PROGRAM_NAME = "pc2sandbox.sh";
     
+    /**
+     * Constants for interactive problems
+     */
+    // Filename of the script to run for an interactive problem in a sandbox
```
Comment (lane55) [inline src/edu/csus/ecs/pc2/core/Constants.java]: Incorrect comment layout, for javadoc and quick lookup in eclipse, use layout
```
/**
 * <describe field>
 */
```

**Review by lane55 — APPROVED** [review]
Minor comments

File: scripts/pc2_interactive.sh (hunk 1 above)
Comment (johnbrvc) [reply scripts/pc2_interactive.sh:89]: Fixed.

File: scripts/pc2sandbox_interactive.sh (hunk 3 above)
Comment (johnbrvc) [reply scripts/pc2sandbox_interactive.sh:114]: Fixed.

File: src/edu/csus/ecs/pc2/core/Constants.java (hunk 4 above)
Comment (johnbrvc) [reply src/edu/csus/ecs/pc2/core/Constants.java]: Fixed.

**Pull request merged**